cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/SQLEngine.java

```diff
@@ -57,19 +59,19 @@
    * After created, this dataset will be considered "locked" until the output has been committed.
    *
    * @param pushRequest the request containing information about the dataset name and schema.t
-   * @return an {@link OutputFormatProvider} instance that can be used to write records to the SQL Engine into the
-   * specified dataset.
+   * @return {@link SQLPushDataset} instance that can be used to write records to the SQL Engine.
    */
-  OutputFormatProvider getPushProvider(SQLPushRequest pushRequest) throws SQLEngineException;
+  SQLPushDataset<StructuredRecord, KEY_OUT, VALUE_OUT> getPushProvider(SQLPushRequest pushRequest)
+    throws SQLEngineException, InterruptedException;
```

Comment:
I don't think InterruptedException is needed here and in other methods. It exposes too much about implementation (that there is a blocking operation going on in this method). Even if there is a blocking operation, it's better to catch and rethrow as SQLEngineException (or it's subclass if you need to be specific).

Reply:
Fair point, although I'm not a huge fan of wrapping all exceptions into only one class.